### PR TITLE
RavenDB-5770 Implement Stream for the new client - WIP

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.NewClient/Commands/StreamCommand .cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        public void SetResponse(Stream response)
+        {
+            //WIP
+            if (response == null)
+                throw new InvalidOperationException();
+            var buffer = new byte[12];
+            response.Read(buffer, 0, 12);
+            //TODO  - better exception
+            if (string.Compare(buffer.ToString(), "{\"Results\":[") != 1)
```
Comment (ayende) [inline src/Raven.NewClient/Commands/StreamCommand .cs:38]: Um? What?! That isn't going to work.
You need to actually parse this properly. 
The slightest change can break this horribly (for example, adding a space.

Leaving aside that `byte[].ToString()` is going to return `System.Byte[]`, not the content of the array.

File: src/Raven.NewClient/Commands/StreamOperation.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+using Raven.NewClient.Client.Data.Queries;
+using Raven.NewClient.Client.Linq;
+using Sparrow.Json;
+
+namespace Raven.NewClient.Client.Commands
+{
+    public class StreamOperation
+    {
+        private readonly InMemoryDocumentSessionOperations _session;
+        private static readonly Logger _logger = LoggingSource.Instance.GetLogger<StreamOperation>("Raven.NewClient.Client");
```
Comment (ayende) [inline src/Raven.NewClient/Commands/StreamOperation.cs:19]: Probably should just be Raven.Client

File: src/Raven.NewClient/Commands/StreamOperation.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+        public StreamCommand CreateRequest<T>(IDocumentQuery<T> query)
+        {
+            var ravenQueryInspector = ((IRavenQueryInspector)query);
+            var indexQuery = ravenQueryInspector.GetIndexQuery(false);
+            if (indexQuery.WaitForNonStaleResults || indexQuery.WaitForNonStaleResultsAsOfNow)
+                throw new NotSupportedException(
+                    "Since Stream() does not wait for indexing (by design), streaming query with WaitForNonStaleResults is not supported.");
+            _session.IncrementRequestCount();
+            var reference = new Reference<QueryHeaderInformation>();
```
Comment (ayende) [inline src/Raven.NewClient/Commands/StreamOperation.cs:39]: Not used

File: src/Raven.NewClient/Commands/StreamOperation.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                Index = path,
+            };
+        }
+
+        public IEnumerator<BlittableJsonReaderObject> SetResult<T>(Stream result)
+        {
+            //WIP
+            using (var parser = new JsonOperationContext.MultiDocumentParser(_session.Context, result))
+            {
+                while (parser._parser.BufferOffset + 2 != parser._parser.BufferSize)
```
Comment (ayende) [inline src/Raven.NewClient/Commands/StreamOperation.cs:64]: Why do it like this, you can just continue until you get to the `EndOfStreamException`

Comment (ayende) [reply src/Raven.NewClient/Commands/StreamOperation.cs:64]: Or change it to actually be valid

File: src/Raven.NewClient/Commands/StreamCommand .cs (hunk 1 above)
Comment (efratshenhar) [reply src/Raven.NewClient/Commands/StreamCommand .cs:38]: yes , i know . this is temporary ( i wrote that it is WIP).

File: src/Raven.NewClient/Commands/StreamOperation.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            var reference = new Reference<QueryHeaderInformation>();
+            var index = ravenQueryInspector.IndexQueried;
+            if (string.IsNullOrEmpty(index))
+                throw new ArgumentException("Key cannot be null or empty index");
+            string path;
+            if (indexQuery.Query != null && indexQuery.Query.Length > _session.Conventions.MaxLengthOfQueryUsingGetUrl)
+            {
+                path = indexQuery.GetIndexQueryUrl(index, "streams/queries", includePageSizeEvenIfNotExplicitlySet: false, includeQuery: false);
+            }
+            else
```
Comment (ayende) [inline src/Raven.NewClient/Commands/StreamOperation.cs:48]: What happens in this code path?

**Review by ayende — COMMENTED** [review]
See my changes on top of yours, it isn't the way to handle it.
You need to actually parse the strings directly

**Pull request merged**